nmmo/task/game_state.py

```diff
@@ -0,0 +1,94 @@
+from typing import Dict, List, Tuple
+from dataclasses import dataclass
+from copy import deepcopy
+
+import numpy as np
+import numpy_indexed as npi
+
+from nmmo.core.config import Config
+from nmmo.core.realm import Realm
+from nmmo.core.observation import Observation
+
+from nmmo.entity.entity import EntityState
+from nmmo.systems.item import ItemState
+from nmmo.core.tile import TileState
+
+
+@dataclass
+class GameState:
+  tick: int
+  config: Config
+  spawn_pos: Dict[int, Tuple[int, int]] # ent_id: (row, col)
+
+  alive_agents: List # of alive agents' ent_id
+  env_obs: Dict[int, Observation] # this should include only alive agents
+
+  entity_cols: Dict # attr_to_col
```

Comment:
Can we just use Entity.State.attr_to_col_name? No point in copying that into every GameState when it can just be imported

Reply:
You can just add 

EntityAttr = EntityState.State.attr_name_to_col

To the top of the file

Reply:
defined EntityAttr, ItemAttr, TileAttr at the top of the file and using these, instead of passing these via game state.